src/common/common.py

```diff
@@ -8,9 +8,65 @@
 from pathlib import Path
 from streamlit.components.v1 import html
 
+# Optional plotting package imports
+try:
+    import plotly.io as pio
+
+    PLOTLY_AVAILABLE = True
+except ImportError:
+    PLOTLY_AVAILABLE = False
+
+try:
+    import matplotlib.pyplot as plt
+
+    MPL_AVAILABLE = True
+except ImportError:
+    MPL_AVAILABLE = False
+
+try:
+    from bokeh.io import curdoc
+
+    BOKEH_AVAILABLE = True
+except ImportError:
+    BOKEH_AVAILABLE = False
+
 import streamlit as st
-import pandas as pd
-import psutil
+
+# Optional pandas import
+try:
```

Comment:
Pandas should be always available. It is just the plotting packages that are not fixed as dependencies so we should only check for them.